formatter/action.yml

```diff
@@ -17,19 +17,24 @@ runs:
   steps:
     - run: echo "${{ github.action_path }}" >> $GITHUB_PATH
       shell: bash
-    - name: Setup Git configurations
-      shell: bash
-      run: |
-        git config --global user.name github-actions[bot]
-        git config --global user.email 'github-actions@users.noreply.github.com'
     - name: Running the formatter
       shell: bash
       run: |
         filename_formatter.sh ${{ inputs.working-directory }} ${{ inputs.filetypes }} ${{ inputs.ignore-files }}
-    - name: Committing changes
+
+        git pull || true
+    - name: Committing and pushing changes
+      uses: stefanzweifel/git-auto-commit-action@v4
+      id: commit-push
+      with:
+        commit_message: "chore: formatting filenames"
+        branch: "formatting_files-${{ github.sha }}"
+        create_branch: true
+    - name: Creating and merging the PR
       shell: bash
+      if: steps.commit-push.outputs.changes_detected == 'true'
       run: |
-        git add ${{ inputs.working-directory }} || true
-        git commit -m "chore: formatting filenames" || true
-
-        git push origin HEAD:$GITHUB_REF || true
+        gh pr create --base ${GITHUB_REF##*/} --head formatting_files-${{ github.sha }} --title 'chore: formatting filenames' --body 'Formatted filenames (see the diff. for changes).'
+        gh pr merge --admin --merge --subject 'chore: formatting filenames' --delete-branch
+      env:
+        GH_TOKEN: ${{ github.token }}
```

Comment:
BTW on a personal level, I really think CI checks should help find out the problems for the contributors not _fix_ them so I think fixing the filenames into the correct format automatically means that the contributor really never learns to read the contribution guidelines 😅 - our check should indicate to them that there is a problem with their PR with a comment or something instead of fixing it for them.

Reply:
The CI actually updates the filenames in case the PR creator didn't follow our guidelines.
In this case, it would create a PR to their fork and update it.

But, sometimes GitHub Actions aren't enabled in forks, which then will be updated in our repository in case anything was missed.

Reply:
So, I have time to get back to this now, hehe.
I think I agree with you. Should I edit it so that the action fails if there's anything found, and specify where the error is?

Reply:
I've been off discord for a while haha, I'll check again soon 😄